### ISSUE TITLE
CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle)

### ISSUE THREAD

**Issue by idelsink** [derpicated/articated#13]
When calling CMake to generate the makefile I get the following error:

```text
CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
  "operators" of type STATIC_LIBRARY
    depends on "movement3d" (weak)
    depends on "operators_autogen" (strong)
  "movement3d" of type STATIC_LIBRARY
    depends on "operators" (weak)
    depends on "movement3d_autogen" (strong)
  "operators_autogen" of type UTILITY
    depends on "movement3d" (strong)
  "movement3d_autogen" of type UTILITY
    depends on "operators" (strong)
At least one of these targets is not a STATIC_LIBRARY.  Cyclic dependencies are allowed only among static libraries.
```

I'm calling CMake as follows:
```sh
cmake .. -DQt5_DIR="/home/ingmar/Qt5.9.1/5.9.1/gcc_64/lib/cmake/Qt5"
```

**Comment by Mercotui** [comment]
You say this is fixed in #14? If this is correct, then please close the issue 👀

**Comment by idelsink** [comment]
It will be automatically closed when it's merged into master.
But sure, I can close it.
